go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -260,7 +276,18 @@ func getDbStates(ctx context.Context, mrEnv env.MultiRepoEnv, dbs []dsqle.Databa
 			return nil, fmt.Errorf("couldn't find environment for database %s", db.Name())
 		}
 
+		if _, val, ok := sql.SystemVariables.GetGlobal(SqlServerBranchRefKey); ok {
```

Comment:
This is definitely not what you want.

This is using the repo state writer in doltEnv to write a new branch to the repo state, but without any of the business logic in `dolt checkout`.

Think of the command line env as a separate, persistent session. You're changing the branch of that session as a side effect to this logic.

You need to add the default branch to the session, and add logic to newSessionBuilder to set it there, initialized from the value of the global sys var at connection time

Reply:
i still don't know how to test this thing for weird cases